client/client.go

```diff
@@ -213,3 +235,25 @@ func WithWatcher(wc WatcherCtor) Option {
 		return nil
 	}
 }
+
+// PrometheusBridge abstracts the Prometheus metric registration and push functionalities.
+type PrometheusBridge interface {
+	Register(prometheus.Collector) error
+	Push() error
+}
+
+// WithPrometheus specifies a prometheus system to be used for metric collection.
+func WithPrometheus(b PrometheusBridge) Option {
+	return func(cfg *clientConfig) error {
+		cfg.prometheus = b
+		return nil
+	}
+}
```

Comment:
can this be simplified to `WithRegistry(prometheus.Registerer)`
https://github.com/prometheus/client_golang/blob/master/prometheus/registry.go#L97